src/SignalR/server/Core/src/DefaultHubLifetimeManager.cs

```diff
@@ -238,7 +238,26 @@ public override Task SendGroupsAsync(IReadOnlyList<string> groupNames, string me
             var group = _groups[groupName];
             if (group != null)
             {
-                DefaultHubLifetimeManager<THub>.SendToGroupConnections(methodName, args, group, null, null, ref tasks, ref message, cancellationToken);
+                foreach (var connection in group)
+                {
+                    if (connections == null)
+                    {
+                        connections = new HashSet<string>();
+                    }
+                    connections.Add(connection.Key);
```

Comment:
FYI this could be written also as 1-liner.
```suggestion
                    (connections ?? []).Add(connection.Key);
```
For me it's readability is the same.

---

In L239 can you also check if the `group` has items?
If so, the check for `connections == null` can be removed altogehter, and initialize the HashSet just before the foreach-loop in L241, because if the group has items, then Add is called unconditionally.

The HashSet could then be also initialized with the proper capacity, to avoid potential resizing and copying of the internal structures of HashSet.